#! /usr/bin/env python

PACKAGE='os5000'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name          Type      Level Description                                    Default Min   Max
gen.add("port",       str_t,    0,    "The port the compass is plugged into.",       "/dev/ttyUSB0")
gen.add("baud",       int_t,    0,    "The baud rate of the compass.",               115200, 2400, 115200)
gen.add("rate",       int_t,    0,    "The rate the compass should output data.",    40,     -50,  40)
gen.add("init_time",  int_t,    0,    "The amount of time to connect to compass.",   1,      1,    10)
gen.add("reconnect",  bool_t,   0,    "Whether to attempt to reconnect to compass.", False)
gen.add("roll_sim",   double_t, 0,    "Simulated roll angle.",                       0.0,    -180.0, 180.0)
gen.add("pitch_sim",  double_t, 0,    "Simulated pitch angle.",                      0.0,    -180.0, 180.0)
gen.add("yaw_sim",    double_t, 0,    "Simulated yaw angle.",                        0.0,    -180.0, 180.0)

exit(gen.generate(PACKAGE, "os5000", "os5000"))
